service/worker/indexer/esProcessor.go

```diff
@@ -287,6 +301,25 @@ func (p *ESProcessorImpl) hashFn(key interface{}) uint32 {
 	return uint32(common.WorkflowIDToHistoryShard(id, numOfShards))
 }
 
+func (p *ESProcessorImpl) isDeleteRequest(request bulk.GenericBulkableRequest) bool {
+	req, err := request.Source()
+	if err != nil {
+		p.logger.Error("Get request source err.", tag.Error(err), tag.ESRequest(request.String()))
+		p.scope.IncCounter(metrics.ESProcessorCorruptedData)
+		return false // don't nack the message, let the processor retry
+	}
+
+	if len(req) == 1 {
+		// The Source() method typically returns a slice of strings, where each string represents a part of the bulk request in JSON format.
+		// For delete operations, the Source() method typically returns only one part
+		// The metadata that specifies the delete action, including _index and _id.
+		// "{\"delete\":{\"_index\":\"my-index\",\"_id\":\"1\"}}"
+		return true
```

Comment:
Is this a future proof way to detect deletion? does index return more than one string?

Reply:
Yeah, this is a generic bulkable request used by different es versions and os. 
For index/updates, there will be more. I see we are also using this way to retrieve kafka key for requests https://github.com/uber/cadence/blob/master/service/worker/indexer/esProcessor.go#L245

Reply:
I see thanks for checking other similar usages. Still sounds like bug-prone but if there are no other better ways for now let's keep the same logic. 
In order to avoid spreading this logic all over let's add a `IsDeleteRequest(request bulk.GenericBulkableRequest) bool` function and call it here and from L245

Reply:
Update! I have found why we are doing the check in this way, we are using the bulk api for all operations, https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-bulk.html and https://opensearch.org/docs/latest/api-reference/document-apis/bulk/.
For all bulk requests, they are []string, for delete, it will be one string containing the index and docID.
For index/update, they have extra string with all serialized parameters.